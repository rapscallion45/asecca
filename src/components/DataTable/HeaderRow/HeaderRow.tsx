import { FC } from 'react';
import { styled } from '@mui/material/styles';
import TableRow from '@mui/material/TableRow';
import TableCell, { tableCellClasses } from '@mui/material/TableCell';
import { IDataTableColumn } from '@/components/DataTable/types';

/**
 * table header cell stylings
 *
 * @since - 0.0.0
 */
const StyledTableHeadCell = styled(TableCell)(({ theme }) => ({
  /** table header colors */
  [`&.${tableCellClasses.head}`]: {
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.primary.contrastText,
    fontSize: 16,
    fontWeight: 400,
  },
}));

/**
 * Data Table Header Row Props
 *
 * @since - 0.0.0
 *
 * @typedef IHeaderRowProps
 * @prop {Array<IDataTableColumn>} columns - data table columns
 */
interface IHeaderRowProps {
  columns: Array<IDataTableColumn>;
}

/**
 * Data Table Header Row
 *
 * @author - [Carl Scrivener](https://github.com/rapscallion45)
 * @since - 0.0.0
 *
 * @param {IHeaderRowProps} props - component props
 * @returns {FC} - data table header row functional component
 */
const HeaderRow: FC<IHeaderRowProps> = (props) => {
  const { columns } = props;

  return (
    <TableRow>
      {/** map passed columns to table headers row */}
      {columns.map((column: IDataTableColumn) => (
        <StyledTableHeadCell key={column.key} align="left">
          {column.label}
        </StyledTableHeadCell>
      ))}
    </TableRow>
  );
};

export default HeaderRow;
