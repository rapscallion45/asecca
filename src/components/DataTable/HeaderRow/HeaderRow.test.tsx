import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Table, TableBody } from '@mui/material';
import HeaderRow from './HeaderRow';
import columnsMock from '../../../../__mocks__/dataTableColumnsMock';

/* Heading Row Unit Tests */
/* ====================== */
describe('Data Table Header Row', () => {
  it('Renders correctly', async () => {
    /* perform snapshot test */
    const tree = renderer
      .create(
        <Table>
          <TableBody>
            <HeaderRow columns={columnsMock} />
          </TableBody>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Message', () => {
    it('Should render the passed message text', async () => {
      /* Arrange */
      /* Act */
      render(
        <Table>
          <TableBody>
            <HeaderRow columns={columnsMock} />
          </TableBody>
        </Table>
      );

      /* Assert - make sure each column heading is rendered */
      columnsMock.forEach((column) => {
        expect(screen.getByText(column.label)).toBeInTheDocument();
      });
    });
  });
});
