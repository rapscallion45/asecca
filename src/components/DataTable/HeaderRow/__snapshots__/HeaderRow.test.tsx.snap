// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Header Row Renders correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Product
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Application
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Global
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Customer
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Project
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Collection
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-hb3prk-MuiTableCell-root"
      >
        Prevailing
      </td>
    </tr>
  </tbody>
</table>
`;
